src/Avalonia.Base/Collections/AvaloniaListExtensions.cs

```diff
@@ -145,7 +147,13 @@ void Remove(int index, IList items)
             var result = new AvaloniaList<TDerived>();
 
             collection.ForEachItem(
-                (i, item) => result.Insert(i, select(item)),
+                (i, item) =>
+                {
```

Comment:
There is one really nasty bug in DevTools, if you press strl+shift over listbox and scroll there is a crash (OutOfRangeException in devtools treeeview ... and app caboom )that's fixed here. if you want i can submit  pr with only this fix

Reply:
actually made separate pr for it, #2060